spark/datadog_checks/spark/spark.py

```diff
@@ -199,79 +205,64 @@ def check(self, instance):
                 message='Connection to ApplicationMaster "%s" was successful' % am_address,
             )
 
-    def _get_master_address(self, instance):
+    def _get_master_address(self):
         """
         Get the master address from the instance configuration
         """
 
-        master_address = instance.get(MASTER_ADDRESS)
+        master_address = self.instance.get(MASTER_ADDRESS)
         if master_address is None:
-            master_address = instance.get(DEPRECATED_MASTER_ADDRESS)
+            master_address = self.instance.get(DEPRECATED_MASTER_ADDRESS)
 
             if master_address:
                 self.log.warning(
                     'The use of `%s` is deprecated. Please use `%s` instead.', DEPRECATED_MASTER_ADDRESS, MASTER_ADDRESS
                 )
             else:
-                raise Exception('URL for `%s` must be specified in the instance configuration' % MASTER_ADDRESS)
+                raise ConfigurationError(
+                    'URL for `%s` must be specified in the instance configuration' % MASTER_ADDRESS
+                )
 
         return master_address
 
-    def _get_request_url(self, instance, url):
+    def _get_request_url(self, url):
         """
         Get the request address, build with proxy if necessary
         """
         parsed = urlparse(url)
 
         _url = url
-        if not (parsed.netloc and parsed.scheme) and is_affirmative(instance.get('spark_proxy_enabled', False)):
-            master_address = self._get_master_address(instance)
-            _url = urljoin(master_address, parsed.path)
+        if not (parsed.netloc and parsed.scheme) and is_affirmative(self.instance.get('spark_proxy_enabled', False)):
+            _url = urljoin(self.master_address, parsed.path)
 
         self.log.debug('Request URL returned: %s', _url)
         return _url
 
-    def _get_running_apps(self, instance):
+    def _get_running_apps(self):
         """
         Determine what mode was specified
         """
-        tags = instance.get('tags', [])
-        if tags is None:
-            tags = []
-        master_address = self._get_master_address(instance)
-        # Get the cluster name from the instance configuration
-        cluster_name = instance.get('cluster_name')
-        if cluster_name is None:
-            raise Exception('The cluster_name must be specified in the instance configuration')
-        tags.append('cluster_name:%s' % cluster_name)
-        tags = list(set(tags))
-        # Determine the cluster mode
-        cluster_mode = instance.get(SPARK_CLUSTER_MODE)
-        if cluster_mode is None:
-            self.log.warning(
-                'The value for `spark_cluster_mode` was not set in the configuration. Defaulting to "%s"',
-                SPARK_YARN_MODE,
-            )
-            cluster_mode = SPARK_YARN_MODE
+        tags = list(self.tags)
```

Comment:
Is `self.tags` not a list to begin with?

Reply:
It's to avoid the memory leak issue we had with elasticsearch fixed with #8213